lib/matplotlib/ticker.py

```diff
@@ -1999,36 +2009,41 @@ def view_limits(self, vmin, vmax):
         'Try to choose the view limits intelligently'
         b = self._base
 
-        if vmax < vmin:
-            vmin, vmax = vmax, vmin
+        vmin, vmax = self.nonsingular(vmin, vmax)
 
         if self.axis.axes.name == 'polar':
             vmax = math.ceil(math.log(vmax) / math.log(b))
             vmin = b ** (vmax - self.numdecs)
-            return vmin, vmax
-
-        minpos = self.axis.get_minpos()
-
-        if minpos <= 0 or not np.isfinite(minpos):
-            raise ValueError(
-                "Data has no positive values, and therefore can not be "
-                "log-scaled.")
-
-        if vmin <= 0:
-            vmin = minpos
 
         if rcParams['axes.autolimit_mode'] == 'round_numbers':
             if not is_decade(vmin, self._base):
                 vmin = decade_down(vmin, self._base)
             if not is_decade(vmax, self._base):
                 vmax = decade_up(vmax, self._base)
 
-            if vmin == vmax:
-                vmin = decade_down(vmin, self._base)
-                vmax = decade_up(vmax, self._base)
+        return vmin, vmax
 
-        result = mtransforms.nonsingular(vmin, vmax)
-        return result
+    def nonsingular(self, vmin, vmax):
```

Comment:
can this have a leading underscore?  I missread and did not notice that this was a method.